scipy/ndimage/_filters.py

```diff
@@ -1491,9 +1492,26 @@ def _rank_filter(input, rank, size=None, footprint=None, output=None,
             raise RuntimeError(
                 "A sequence of modes is not supported by non-separable rank "
                 "filters")
-        mode = _ni_support._extend_mode_to_code(mode)
-        _nd_image.rank_filter(input, rank, footprint, output, mode, cval,
-                              origins)
+        mode = _ni_support._extend_mode_to_code(mode, is_filter=True)
+        if input.ndim == 1:
+            if input.dtype in (np.int64, np.float64, np.float32):
+                x = input
+                x_out = output
+            elif input.dtype == np.float16:
+                x = input.astype('float32')
+                x_out = np.empty(x, dtype='float32')
+            elif np.result_type(input, np.int64) == np.int64:
+                x = input.astype('int64')
+                x_out = np.empty(x, dtype='int64')
```

Comment:
```suggestion
                x = input.astype('float32')
                x_out = np.empty(x.shape, dtype='float32')
            elif np.result_type(input, np.int64) == np.int64:
                x = input.astype('int64')
                x_out = np.empty(x.shape, dtype='int64')
```

Reply:
Thanks for this MR, @ggkogan. Please see if this suggestion resolves the test failures

Reply:
The bug was using `x` instead of `x.shape` as the first argument to `np.empty`.

I also changed the integer case to also work for bool and uint64. Please update an existing test case as follows to test across the dtypes:
```py

    @pytest.mark.parametrize(
        'dtype',
        [np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32,
         np.uint64, bool, np.float16, np.float32, np.float64],
    )
    def test_rank02(self, dtype):
        array = np.array([1, 2, 3, 4, 5], dtype=dtype)
        output = ndimage.rank_filter(array, 1, size=[3])
        assert_array_almost_equal(array, output)
        output = ndimage.percentile_filter(array, 50, size=3)
        assert_array_almost_equal(array, output)
        output = ndimage.median_filter(array, (3,))
        assert_array_almost_equal(array, output)
```